core/src/test/scala/org/apache/spark/util/collection/ExternalSorterSuite.scala

```diff
@@ -506,7 +506,10 @@ class ExternalSorterSuite extends FunSuite with LocalSparkContext with PrivateMe
     val agg = new Aggregator[Int, Int, Int](i => i, (i, j) => i + j, (i, j) => i + j)
     val ord = implicitly[Ordering[Int]]
     val sorter = new ExternalSorter(Some(agg), Some(new HashPartitioner(3)), Some(ord), None)
-    sorter.insertAll((0 until 100000).iterator.map(i => (i / 2, i)))
+
+    // avoid combine before spill
+    sorter.insertAll((0 until 50000).iterator.map(i => (i , 2 * i)))
+    sorter.insertAll((0 until 50000).iterator.map(i => (i, 2 * i + 1)))
```

Comment:
Yes, it failed on my computer(this testcase didn't stop). I set the memory limit to a small value  on my computer to ganrantee spills. I think we can also set `spark.shuffle.memoryFraction` to a small value to do this. The origin value of `spark.shuffle.memoryFraction` is 0.001, I don't know if it can ganrantee spills. If not, we should change it to a smaller value.